engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/PartSubGraphCompiler.scala

```diff
@@ -147,13 +149,21 @@ class PartSubGraphCompiler(protected val expressionCompiler: ExpressionCompiler,
         fa = compile(next, ctx))(
         f = compiledNext => compiledgraph.node.CustomNode(id, compiledNext))
 
-    case SubprocessInput(id, ref, _, _) =>
+    case SubprocessInput(id, ref, _, _, subParams) =>
       val childCtx = ctx.pushNewContext(globalVariableTypes)
-
       val newCtx = ref.parameters.foldLeft[ValidatedNel[ProcessCompilationError, ValidationContext]](Valid(childCtx))
                     { case (accCtx, param) => accCtx.andThen(_.withVariable(param.name, Unknown))}
-      val validParams =
-        expressionCompiler.compileObjectParameters(ref.parameters.map(p => Parameter.unknownType(p.name)), ref.parameters, toOption(ctx))
+
+      val getSubprocessParam = (paramName: String) => {
+        val subParam = subParams.get.find(_.name == paramName).get
+        val typingR = subParam.typ.toTyped(classLoader).getOrElse(throw new IllegalArgumentException("Failed to load subprocess parameter type"))
```

Comment:
maybe some Invalid instead of exception?

Reply:
good idea